### ISSUE TITLE
Renaming (F2) parameter in function declaration in header file doesn't rename it in defintion

### ISSUE THREAD

**Issue by Spaark84** [microsoft/vscode-cpptools#9746]
### Environment

- OS and Version: Windows 10
- VS Code Version: 1.70.2
- C/C++ Extension Version: 1.11.5

### Bug Summary and Steps to Reproduce

When renaming (F2) a parameter in a function declaration in a header file, the parameter in the function definition in the source file that includes the header file is not renamed and it is also not listed in the refactor preview. The same applies vice versa.

However, if the function declaration is placed in the source file, the parameter in the function definition is listed in the refactor preview if the parameter in the declaration is renamed and vice versa.

### Expected behavior

The parameter in the function definition should be listed in the refactor preview for selection if the parameter in the declaration is renamed and vice versa, regardless of whether the declaration is placed in the source or in a included header file.

### Code sample and Logs

```shell
// test.c

#include "test.h"

void foo(int arg)
{
  return;
}


// test.h

void foo(int arg);
```

### Screenshots

Refactor preview when renaming a parameter in the **declaration** when it is placed in the source file:

![image](https://user-images.githubusercontent.com/58564478/185380337-2692c64e-f845-4749-b6a9-58501fbffd6d.png)

Refactor preview when renaming a parameter in the **definition** when it is placed in the source file:

![image](https://user-images.githubusercontent.com/58564478/185380535-db3bf5ca-77e2-40df-95b3-f62dfd9f8133.png)

After renaming a parameter in the **declaration** when it is placed in the included header file (no refactor preview):

![image](https://user-images.githubusercontent.com/58564478/185380954-7d7e2191-47de-40a4-8dad-b646e4f45f23.png)

After renaming a parameter in the **definition** when it is placed in the included header file (no refactor preview):

![image](https://user-images.githubusercontent.com/58564478/185380811-3a33227b-1dfa-4c64-b490-9c96d8db1b73.png)

**Comment by Spaark84** [comment]
@sean-mcmanus, you are absolutely right. Still, I would find it useful if the function parameter of the related declaration or definition would be listed in the refactor view if the name of the parameter is the same, but with a unchecked checkbox by default, as is the case when the declaration is placed in the source file.

**Comment by sean-mcmanus** [comment]
Oh, I didn't realize the rename worked with the declaration in the source file -- it doesn't work with VS 2022. I've filed a bug on VS at https://developercommunity.visualstudio.com/t/C-IntelliSense-rename-of-a-function-de/10126023 .

**Comment by bobbrow** [comment]
> Oh, I didn't realize the rename worked with the declaration in the source file

Well... it doesn't work in the sense that it appears as "not a match".  It works in the sense that you can override our default behavior by checking the box to include it in the rename operation.

**Comment by sean-mcmanus** [comment]
@bobbrow Yeah, that's what I meant (found as a non-confirmed reference).
